tests/codemods/test_combine_startswith_endswith.py

```diff
@@ -52,3 +53,170 @@ def test_exclude_line(self, tmpdir):
+        self._format_func_run_test(tmpdir, func, input_code, expected)
+
+    @each_func
+    @pytest.mark.parametrize(
+        "input_code, expected",
+        [
+            # cst.ConcatenatedString on the left
+            (
+                "x.{func}('foo' 'bar') or x.{func}('baz')",
+                "x.{func}(('foo' 'bar', 'baz'))",
+            ),
+            # cst.ConcatenatedString on the right
+            (
+                "x.{func}('foo') or x.{func}('bar' 'baz')",
+                "x.{func}(('foo', 'bar' 'baz'))",
+            ),
+            # cst.ConcatenatedString on both sides
+            (
+                "x.{func}('foo' 'bar') or x.{func}('baz' 'qux')",
+                "x.{func}(('foo' 'bar', 'baz' 'qux'))",
+            ),
+        ],
+    )
+    def test_concat_string_args(self, tmpdir, func, input_code, expected):
+        self._format_func_run_test(tmpdir, func, input_code, expected)
+
+    @each_func
+    @pytest.mark.parametrize(
+        "input_code, expected",
+        [
+            # cst.FormattedString on the left
+            (
+                "x.{func}(f'formatted {foo}') or x.{func}('bar')",
+                "x.{func}((f'formatted {foo}', 'bar'))",
+            ),
+            # cst.FormattedString on the right
+            (
+                "x.{func}('foo') or x.{func}(f'formatted {bar}')",
+                "x.{func}(('foo', f'formatted {bar}'))",
+            ),
+            # cst.FormattedString on both sides
+            (
+                "x.{func}(f'formatted {foo}') or x.{func}(f'formatted {bar}')",
+                "x.{func}((f'formatted {foo}', f'formatted {bar}'))",
+            ),
+        ],
+    )
+    def test_format_string_args(self, tmpdir, func, input_code, expected):
+        self._format_func_run_test(tmpdir, func, input_code, expected)
+
+    @each_func
+    @pytest.mark.parametrize(
+        "input_code, expected",
+        [
+            # cst.Name on the left
+            ("x.{func}(y) or x.{func}('foo')", "x.{func}((y, 'foo'))"),
+            # cst.Name on the right
+            ("x.{func}('foo') or x.{func}(y)", "x.{func}(('foo', y))"),
+            # cst.Name on both sides
+            ("x.{func}(y) or x.{func}(z)", "x.{func}((y, z))"),
+            # cst.Name in tuple on the left
+            ("x.{func}((y, 'foo')) or x.{func}('bar')", "x.{func}((y, 'foo', 'bar'))"),
+            # cst.Name in tuple on the right
+            ("x.{func}('foo') or x.{func}((y, 'bar'))", "x.{func}(('foo', y, 'bar'))"),
+            # cst.Name in tuple on both sides
+            (
+                "x.{func}((y, 'foo')) or x.{func}((z, 'bar'))",
+                "x.{func}((y, 'foo', z, 'bar'))",
+            ),
+        ],
+    )
+    def test_name_args(self, tmpdir, func, input_code, expected):
+        self._format_func_run_test(tmpdir, func, input_code, expected)
+
+    _100_of_each_type = [
+        (f"'{i}'", f"'{i}con' 'cat{i}'", f"f'fmt{i}'", f"name{i}") for i in range(100)
```

Comment:
I gotta ask, any reason we need 100? Seems like if you test with 3 or 4 that should prove the point that we can support Nth types.

Reply:
No, you're right this isn't strictly necessary was just getting familiar with the framework when I wrote this so I wanted to make sure I wasn't missing something by testing with a large number.